execution/evm/vm_test.go

```diff
@@ -709,12 +717,69 @@ func TestSendCall(t *testing.T) {
 	assert.NoError(t, err, "Expected insufficient gas error")
 }
 
+// Test to ensure that contracts called with STATICCALL cannot modify state
+func TestStaticCall(t *testing.T) {
+	fmt.Printf("%s ---\n", t.Name())
+	inOff, inSize := byte(0x0), byte(0x0) // no call data
+	retOff, retSize := byte(0x0), byte(0x2)
+
+	for _, illegalOp := range []OpCode{SSTORE, LOG0, LOG1, LOG2, LOG3, LOG4, CREATE, CREATE2, SELFDESTRUCT} {
+		cache := state.NewCache(newAppState())
+		ourVm := NewVM(newParams(), crypto.ZeroAddress, nil, logger)
+
+		calleeAccount, calleeAddress := makeAccountWithCode(cache, "callee",
+			MustSplice(illegalOp, PUSH1, 0x1, return1()))
```

Comment:
This in general is not going to be legal solidity. All of the illegal ops take some number of arguments - you'll need to push them onto the stack somewhere otherwise we will just end up with StackUnderflow errors when we try to pop from the stack.

We can fix this by converting your illegal op words into well formed fragments replacing line 726 with something like:

```go
for _, illegalContractCode := range [][]byte{MustSplice(..., SSTORE), ...}
```

Reply:
Flag checking is done in the first instance for each illegal op's case which automatically returns, so afaik superficially checking state modification ops shouldn't require setting specific stack values. Though I agree its probably better to right legal solidity in the test case.